### ISSUE TITLE
Discordification of Riot (feature set request)

### ISSUE THREAD

**Issue by DonaldTsang** [vector-im/riot-web#7487]
- [ ] Room feature support
    - [ ] Custom server-based and room-based emojis
    - [ ] Emoji collection sharing
    - [ ] Emoji reaction for texts and files
    - [ ] Automated (non-manual) key sharing
- [ ] Community support
    - [ ] Community id generation
    - [ ] Community room mass generation
    - [ ] Community room searching
- [ ] Role and access support
    - [ ] Mass room invitation by role or community
    - [ ] Mass room banning/kicking by role or community
    - [ ] Community room sub-categorization and access
    - [ ] Role application and removal
    - [ ] Read-only/Announcement rooms
- [ ] audio chat support
    - [ ] self-host SIP room support (tying SIP with server)
    - [ ] self-host  Mumble room support (tying Mumble with server)
    - [ ] Radio/Playlist bot support
- [ ] collaborative support
    - [ ] self-host Etherpad/Firepad support (tying Mumble with server)
    - [ ] self-host Hackmd support (tying Mumble with server)

**Comment by ptman** [comment]
At least I have the option to search multiple rooms in riot

**Comment by ptman** [comment]
Are emoji collections the same thing as sticker packs?

**Comment by ptman** [comment]
Etherpad is available as a widget

**Comment by DonaldTsang** [comment]
@ptman I would say smoji collections are like stickerpack but hosted differently. It should be more fluid, room or server-specific and easy to trade (as a standard file), and also add (if you are a mod/admin)

**Comment by ptman** [comment]
Reactions: https://github.com/vector-im/riot-web/issues/1357

**Comment by DonaldTsang** [comment]
@ptman that is good progress... but for the other suggestions those features are equally important.

**Comment by dali99** [comment]
> Custom server-based and room-based emojis

https://github.com/matrix-org/matrix-doc/issues/1256

> Emoji reaction for texts and files (useful for voting)

https://github.com/matrix-org/matrix-doc/issues/441

> Automated (non-manual) key sharing

Automating key verification removes the entire point of E2E.
https://github.com/matrix-org/matrix-doc/issues/1267
and Cross Signing: https://github.com/matrix-org/matrix-doc/issues/1680
will make it a lot simpler.

> Multi-room search

???

> Community support

Unsure what you mean here but communities need an overhaul. But here's an interesting issue:
https://github.com/matrix-org/matrix-doc/issues/1215

> audio chat support

Jitsi is supported in client right now. Could look a bit better but it works.
>> self-host Mumble room support (tying Mumble with server)

Nothing stops you from hosting a mumble server and using the mumble web UI as a widget.

> Radio/Playlist bot support

You could again just make a radio website and have it as a widget.. Or write a jitsi bot, or a mumble bot or what you want to do.

>self-host Etherpad/Firepad support (tying *pad with server)
self-host Hackmd support (tying Hackmd with server)

Both of these are also widgettable. And you can just add them to the integration manager. Just like you can self host jitsi through dimension.

**Comment by DonaldTsang** [comment]
@dali99 
> Automating key verification removes the entire point of E2E.
No it does not when there is a group leader that approves who is within the group.
Cross-signing do sound innovative in its own right.
> ???
Multi-room search (the ability to search all rooms you are in) is a core feature of Discord.
> Community Overhaul
The fact still stands that there are certain features within Discord "servers" that needs to applied to Matrix, and it will be good to list them all
> Jitsi is supported in client right now.
But it is really bad in mobile
> Nothing stops you from hosting a mumble server and using the mumble web UI as a widget.
Please make Mumble and/or SIP as a default widget as their apps works in mobile better.
> Both of these are also widgettable. And you can just add them to the integration manager. Just like you can self host jitsi through dimension.
Another one of those times standardization makes things better (at least for those who would like to avoid companies like Google Docs in favor of FOSS hosting)

**Comment by dali99** [comment]
> @dali99

> > Automating key verification removes the entire point of E2E.
> 
> No it does not when there is a group leader that approves who is within the group.
> Cross-signing do sound innovative in its own right.

If you don't verify each other keys there is no way to know who wrote the key, or to who you are encryting the message to. A web of trust thing might be neat, but it still requires some amount of verification.
> > ???
> 
> Multi-room search (the ability to search all rooms you are in) is a core feature of Discord.

So to make it clear: searching all the rooms you're in for some sort of string? That's not a bad idea.
> > Community Overhaul
>
> The fact still stands that there are certain features within Discord "servers" that needs to applied to Matrix, and it will be good to list them all
A lot of these features are already listed, and its important to realize that matrix _isn't_ discord
> > Jitsi is supported in client right now.
>
> But it is really bad in mobile

Right, but it's there. 
> > Nothing stops you from hosting a mumble server and using the mumble web UI as a widget.
> Please make Mumble and/or SIP as a default widget as their apps works in mobile better.
No? You can make mumble or whatever you mean by "SIP" default in your client when people hit the call button. Other people can use what 
> > Both of these are also widgettable. And you can just add them to the integration manager. Just like > > you can self host jitsi through dimension.
>
> Another one of those times standardization makes things better (at least for those who would like to avoid companies like Google Docs in favor of FOSS hosting)

Google docs is just a default widget in Vector-im's hosted proprietary integration manager which they use on riot.im. When you self-host riot, you will use your own integration manager like for example your self hosted dimension to customize what your default widgets will be. That is the matrix standard way to do this.

Using riot.im you can also just use the custom widget button to add your stuff. Don't confuse riot.im with matrix :smiley:

**Comment by Half-Shot** [comment]
>>Multi-room search (the ability to search all rooms you are in) is a core feature of Discord.

>So to make it clear: searching all the rooms you're in for some sort of string? That's not a bad idea.

![image](https://user-images.githubusercontent.com/2072976/46903025-14592380-cec7-11e8-85a0-2a67ba595aca.png)

**Comment by dali99** [comment]
> > If you don't verify each other keys there is no way to know who wrote the key, or to who you are encrypting the message to. A web of trust thing might be neat, but it still requires some amount of verification.
> 
> In private Discord servers, people are generally invited by the mod/admin to guarantee trust, and if that gets compromised, they move. In Matrix/Riot constant key-signing becomes annoying, so people would rather prefer Discord.
You obviously don't understand how cryptography works, what you are describing is NOT e2e, so just don't use e2e if you don't want to use e2e. A web of trust system again could be neat. but SOME verification must be done.

**Comment by DonaldTsang** [comment]
@dali99 I would say it is a different kind of E2E with a "web of trust" based key distribution.

**Comment by DonaldTsang** [comment]
@Half-Shot another case of bad UI design, since when you want to search all rooms the button is hard to find

**Comment by ptman** [comment]
@DonaldTsang you could give feedback on the UI in https://matrix.to/#/!XIydWLOEJvcDBKwWKE:matrix.org , but the fact is that the functionality already exists

**Comment by DonaldTsang** [comment]
Newer issues for Dscordification:
* https://github.com/vector-im/riot-web/issues/8915
* https://github.com/vector-im/riot-web/issues/8904
* https://github.com/vector-im/riot-web/issues/8873
* https://github.com/vector-im/riot-web/issues/8718
* https://github.com/vector-im/riot-web/issues/8666

**Comment by skylord123** [comment]
@DonaldTsang I've been using Jitsi for voice calling my wife and it works like a charm on mobile. They will be adding push-to-talk to riot soon as well ( #5993 ). So I think it's fine.

**Comment by DonaldTsang** [comment]
@skylord123 Jitsi is not good on mobile though, it is very inconvenient due to its lag and resource usage.

**Comment by DonaldTsang** [comment]
Half-annual report: Discord keeps getting better and better, at this point we should start looking at the RocketChat code base to see how we can make our software more "people-friendly"

**Comment by trosel** [comment]
@skylord123 The Jitsi mobile app is ok. But the Jitsi plugin for Riot on mobile is highly janky.

**Comment by tmladek** [comment]
>Tag-based user and mod/admins types (similar to Discord)
...
> Role and access support 
...
>Community room sub-categorization and access
>>for visitor vs club member vs club staff separation

I was looking for an issue for this; not because I want Matrix to be more like Discord, but because I think it makes sense especially given how often Matrix rooms are bridged right now, and separating "virtual" bot users from other services and actual Matrix users would go a long way to preventing clutter and confusion.

**Comment by DonaldTsang** [comment]
@trosel agreed.

@tmladek I would say that people have put a lot of work to make Discord usable compared to Matrix (since they have more funding)

**Comment by Viohl** [comment]
So, end of 2019. Any updates on an analogue to discords servers?

**Comment by trosel** [comment]
@Viohl That's communities. You can use them today, but the plan is to improve them in the future.

**Comment by DonaldTsang** [comment]
@trosel communities needs to improve UX wise and moderation wise, currently Matrix/Riot is a nightmare to use as a IM alternative to large-scale forum groups.

**Comment by natrius** [comment]
Btw, Emoji-Reactions to text are already there. Would love to see the mumble-support, because we could use it then for some gaming communities. Since mumble is slowly getting plugin-read as well, some communities from arma might switch to mumble as well and it would be awesome  to have it integrated nice in matrix/synapse/dendrite (even riot?) for communication.

**Comment by hramrach** [comment]
mumble is quite nice. Better audio quality than all that software telephony and web conferencing tools. The problem is you do need a central mumble server through which all audio passes if I understand the architecture right. That could be quite a bit of bandwidth for servers with lots of users. Even the message data alone can overload matrix.org. P2P solution on the other hand are prone to connectivity issues if your users are behind firewall.

**Comment by skylord123** [comment]
> P2P solution on the other hand are prone to connectivity issues if your users are behind firewall.

P2P is also prone to people be able to get your IP address. For a while it was pretty common for gamers to add you on skype then DDOS you during games so you would lose (as well as ddos people for other reasons).

**Comment by nathanwfranke** [comment]
New to Riot (from DiscordApp)

It's weird that rooms and communities are separate. I think that if a room is not part of any community it should always be put in the "Direct Messages" dropdown. The rooms dropdown should only be visible if you select a community.

**Comment by DonaldTsang** [comment]
@nathanwfranke here is the problem with that idea: Direct Messages, Rooms that have no community, and Rooms that have community are three things entirely. Rooms that have no community can be converted to have a community, which brings flexibility.

**Comment by nathanwfranke** [comment]
@DonaldTsang This issue is about user unfriendliness. Is it worth it to make the system more "flexible" if it means completely ignoring ease of use?

I understand that flexibility is very important. Maybe then there should be an option to move a Room to a community. But forcing this on every user to understand how all of this is supposed to work is silly.

**Comment by tmladek** [comment]
I would tend to agree - the fact that communities are more or less an optional second-class feature more akin to a kind of metadata, rather than something with proper "weight" like a Discord/Slack server, is not too intuitive and brings confusion. I had quite the trouble explaining this concept even to people who weren't too familiar with (and hence biased by) Discord. (Let alone to people who are, and thus expect Riot Communities and Rooms to have a similar hierarchical relationship.)

I understand the rationale for this, and I also see how it flows naturally from the way rooms are implemented. And I do not advocate for crippling Riot by removing features. However, UI that communicates its purpose well, and nudges the user in the right direction goes a long way, and there needs to be a serious rethinking of how Communities should function within Riot. For instance, clearly marking what Community does a Room belong to (say, in its header) and also marking that a Room does not belong to a Community could help with this confusion already.

Or take hints from how Telegram implemented "Chat Folders" recently, which are actually much closer to how Communities work in Riot (except they're private and cannot be shared) than in Discord - there's an "All Chats" group by default, which looks just the same as the app used to look up until now, and then an arbitrary number of groups which act as filters. https://telegram.org/blog/folders This also communicates quite clearly what kind of relationship do groups and folders (communities) have.

**Comment by natrius** [comment]
I don't get how this is 'complicated'? Put 5 rooms in a box and then you have a collection. The box is a community, just a collection of rooms. It just gets a bit complicated when you have to explain a room can be in 2 communities at the same time. "A community is just a collections of rooms.". 

Or I'm misunderstanding something here.

**Comment by HansJK** [comment]
Most people that use communities are going to look at it as a Discord server or Slack organisation.  So they get confused when they realise that it's just a simple folder.
You can talk about how the current communities UX should have been designed, but the right way forward is to just redesign the whole community feature aka communities 2.0, where it have all the features we have come to expect from Discord and Slack. This includes differentiating the rooms that belongs to a community, more flexible and powerful moderations tool that's role based, and have the member list be based on the community instead of each individual room.

**Comment by DonaldTsang** [comment]
Agreed, Riot2020 (yeah that is what I am nicknaming it) should aim their goals directly at Discord and Slack... in both Speed and Usability (since people have complained that Synapse cannot hold much weight).

I would also like @Half-Shot to comment on the issue with fresh eyes.

**Comment by natrius** [comment]
No, no, no, really not. This should not be a simple discord or slack clone. Communities are already getting a big faceover and we will see if it will be in RiotX for example. https://github.com/matrix-org/matrix-doc/pull/1772

**Comment by HansJK** [comment]
We are not talking about a simple Discord and Slack clone, in fact if we were to completely copy Discord's servers, communities would become much more advanced and usable.
We should take the aspects of these platforms that people love about them and made them so big and popular to begin with. Of course there's many thing Riot could do better, like having the ability to merge communities, or move one community room/channel to another community. A room could even be shared across two or more communities.

I have a fairly large Discord server that's bridged to a Matrix community. When moderating these two platforms, I really see how much more advanced Discord servers are than Matrix communities. I would like to have the best of both worlds.

**Comment by DonaldTsang** [comment]
@natrius @HansJK not clone of Discord just because, but rather a Discord+

> We should take the aspects of these platforms that people love about them and made them so big and popular to begin with

Which is a good way of putting it when I have the concept of Riot2020 or Discord+

> Of course there's many thing Riot could do better, like having the ability to merge communities, or move one community room/channel to another community. A room could even be shared across two or more communities.

Which is useful, and someone that make Discord users feel even more comfortable when using Riot.

>  communities would become much more advanced and usable

That is the crux of the issue, the "server" paradigm of Discord and Slack should be seen as something we derive from, since a lot of people rely on it for moderation... that is not to say it is perfect, but Riot should make moderation even easier than Discord in order to beat it through the flexible "community" paradigm (with the exception where the complexity drive off potential users). Hiding advanced features from people is key to UX I think.

**Comment by Yajo** [comment]
Hi. I'm new to Matrix. I don't use Slack or Discord. I'm just evaluating Matrix to be used as a company communications tool.

What would I expect in a community?

- All my company members should belong to it.
- Admin should choose if anyone can join or they must be invited, and who can invite.
- Notifications for a community could be different from others. For example, you might want for all messages in your working hours, but just mentions outside of them.
- I should be able to send a direct message to anybody in the community by quickly browsing the community members directory.
- There could be volatile rooms or permanent ones.
- People should be able to join or leave some rooms.
- Admin should be able to require encryption community-wide.

I don't know if Slack, Discord or Matrix support these because, as I said, I'm no active user in any of them. This is just what I'd expect.

IMHO if today a Community is just a collection of rooms, then maybe it should be called "folder" or "collection". Maybe Riot could just relabel that option without changing the Collection name in the API, and maybe we could add a new term "Teams" instead of creating a new version of collections.

Just some ideas...

**Comment by ptman** [comment]
@Yajo communities will be reworked in the (hopefully near) future. Probably as soon as e2ee work reaches the next milestone this month. https://github.com/matrix-org/matrix-doc/pull/1772

**Comment by HansJK** [comment]
> What would I expect in a community?
> * People should be able to join or leave some rooms.
> * Admin should be able to require encryption community-wide.
@Yajo The only thing that discord dosen't have is this two. Though you can lave a channel in Slack. Being able to leave a channel is fine, but the community owner should be able to choose whether the members automatically join all rooms or if they have to manually choose what rooms they want to join.

**Comment by squarepupil** [comment]
> 
> 
> @Yajo communities will be reworked in the (hopefully near) future. Probably as soon as e2ee work reaches the next milestone this month. [matrix-org/matrix-doc#1772](https://github.com/matrix-org/matrix-doc/pull/1772)

Assuming I'm understanding the proposal correctly, I see the reworking of communities/groups to be rooms of rooms (gotta love recursion where possible) as being an extremely important step in the direction of "Discordifying" Matrix. If rooms are the fundamental building block of Matrix, then establishing some feature to group them arbitrarily and at-will is what will allow for people to actually build with them. In other words, to create and maintain communities.

I think that this would be "Discordifying" Matrix because Discord's organizing principle of the "server" does many of those things already; "servers" are virtual collections of "channels" (which are roughly equivalent to rooms) that are spun up on demand and which are easily abandoned. Having said this, I don't mean to paint Discord as some paragon of community. On the contrary, Discord "servers", though entirely virtual, are specified like real hardware servers in that they and their contents are stuck where they were placed, which is an extremely limiting design.

Matrix's decentralized mode is much more exciting because a Matrix "community" has the potential to be as fluid as real life communities are. A "community" could contain rooms that originated on dozens of different homeservers, that are accessible simultaneously in multiple communities, and that were cobbled together ad-hoc. This allows the community to figure out what it is _as it's being built_ and to fragment, unify, consolidate, and just do many more of the real life social verbs in general. And all this (hopefully) ad infinitum.

One of the features that Discord implements that I think is crucial to Matrix one day supplanting it in terms of flexibility AND usability (and that @DonaldTsang mentioned in the feature request) is arbitrary role-based administration and user privilege assignment. A linear privilege system like power levels is by no means flexible enough to capture the complexity of needs in even a modestly sized community (e.g., 10 people). It's also much more efficient in allowing for mass actions like mentioning by role (like Discord can do). It also makes semantic sense for communities to establish their own rules by putting names to roles and privileges, which is something that they implicitly (or explicitly) do anyway. This dovetails nicely with the proposal to make communities into rooms of rooms because you could organize roles on a community level. It would need to be worked out how to reconcile conflicting privileges that might exist if the user has particular privileges in rooms contained within a community, but I'm sure this could be done productively. I haven't been able to find a discussion that touches on this in the matrix-doc repository, but I'll do some more digging.

**Comment by SimmyD** [comment]
It would be really amazing if a Community was able to pick a mumble instance which will serve it.  Then under the community you could pull through the voice channels in mumble and integrate in the riot app so you can drop into the channels all in one place.

**Comment by trosel** [comment]
@squarepupil A "room of rooms" would be a house. Call them Houses 🏡 😄 

Each House can have one or many rooms. This is something my grandma could understand.

@SimmyD I know mumble is super high quality and efficient, but the requirement of centralization on mumble is definitely a downside. Kind of antithetical to what matrix is trying to do. 

Something like the original Skype where there is multi-casting going on in the audio room would be ideal. I know Jitsi is like a matrix partner right now, but something more in the direction of [meething](https://github.com/meething/meething) may be better still

**Comment by nathanfranke** [comment]
It's less of a "room of rooms" and more of a collection of rooms. Call them folders or collections. (You can't have a room in multiple houses in that analogy)

**Comment by natrius** [comment]
Or... "Community"? Sorry, could not resist. I'll lead myself out.

**Comment by trosel** [comment]
@nathanfranke 
> It's less of a "room of rooms" and more of a collection of rooms. Call them folders or collections. (You can't have a room in multiple houses in that analogy)

My understanding of @squarepupil is that they're going to re-design how communities work. I agree that communities today are very strange and not-intuitive. Part of the reason for this is the idea that a room can exist in multiple communities. It's... messy.

If a room can exist in multiple "houses", then nothing has changed with the way that communities work today and @squarepupil's [comment here is pointless](https://github.com/vector-im/riot-web/issues/7487#issuecomment-615554409).

I'm sure I'm misunderstanding though.

**Comment by natrius** [comment]
But it should be possible, to be completely honest. There is right now a community for german rooms. He is just collecting german rooms. There can be a community for tech-rooms as well. And a german tech-room can be in both communities at the same time. I think thats really good. The thing is, a room 'can' be in multiple communities. So, why should this be messy. You guys are just not used to that. I like it. 

A clan could create a community for the clan with multiple rooms. And the entry-room can be in the "GAME"-community as well, if desired. 

And in fact, i like that and this should be kept in my opinion, because its great. You CAN use it this way, but you don't have to. Maybe add an option to make it not possible for others to add this room to a community. 

You want it Discord-Like and keep your room in one house/community/whatever-you-want-to-call-it? Fine, just don't add it somewhere else. But keep the functionality for everybody who would like to use it.

**Comment by nathanfranke** [comment]
@natrius I agree that compatibility should be preserved. Many comments here suggest renaming things that aren't beginner-friendly

**Comment by WorldCodeCentral** [comment]
There is a repo for a Mumble web client with active development here https://github.com/Johni0702/mumble-web

Seems to work although sometimes people sound like robots due to audio problems.

Hope that's helpful (I hate using Github).

**Comment by hramrach** [comment]
Is mumble encrypted?

**Comment by natrius** [comment]
A simple google-search shows the following https://wiki.mumble.info/wiki/Features#Encryption 

Is that what you are searching?

**Comment by hramrach** [comment]
There is clearly a conflict in understanding 'communities' or 'servers' vs groups of rooms.

When room belongs to exactly one 'house', 'server', or 'community' there is mapping between the room and the 'community', and you can use it to impose administration and moderation policies on multiple rooms at once.

When 'communities' are folders where anyone can freely file rooms this makes for better organization of rooms but loses this ability to group rooms for purpose of administration/moderation.

**Comment by Mathias-g** [comment]
> There is clearly a conflict in understanding 'communities' or 'servers' vs groups of rooms.
> 
> When room belongs to exactly one 'house', 'server', or 'community' there is mapping between the room and the 'community', and you can use it to impose administration and moderation policies on multiple rooms at once.
> 
> When 'communities' are folders where anyone can freely file rooms this makes for better organization of rooms but loses this ability to group rooms for purpose of administration/moderation.

Yeah, it would be better to call what is currently called communities (A folder of rooms), a List instead. While the new feature on the roadmap, Rooms of Rooms, a group or community.

**Comment by beardedlinuxgeek** [comment]
> There is a repo for a Mumble web client with active development here https://github.com/Johni0702/mumble-web

Unfortunately that broke recently. There is no way to integrate Mumble with Element at this time.

As for Mumble being centralized, so is Jitsi. You need a Jitsi server or you use the central one hosted by vector.

What I would like to see is a tightly coupled integration with Mumble. Make it so Mumble rooms appear as rooms inside of Element instead of using a widget. In Element, on the left hand side, you have a list of "People" rooms and list of "Rooms" rooms. I would like to see a 3rd category which is a list of Mumble channels. Then you could see who is active in the voice channel and join it, like Discord. Basically a Mumble widget which shows in your side bar and not as a widget in the room. Having voice chat is a room widget system is very clunky and the way Discord has it is much nicer.

**Comment by ShadowJonathan** [comment]
I'm not sure if this is related right here, but coming from discord, some UX is incredibly unintuitive for me:
- clicking a user's username makes a mention to their username appear in the chatbox, i expect their profile to pop up
  (the only way i can do that is by scrolling through the room's userlist, which is incredibly user-unfriendly)
- no way to search for existing communities anywhere
- no quickshifter anywhere like discord's ctrl-T/ctrl-K which brings up an universal searcher (which would be perfect for matrix's `+`, `!`, `#` and `@` prefixes for objects
- direct messages and actual rooms are loosely differentiated, and honestly i think the divide is forced, give the user a way to provide:
  - an universal room "list" with all rooms a user is in
  - a "seperated" room list, like "group chat" (room with multiple users but no local `#` ID), "DM" (one recipient, name isnt defined, avatar isnt defined, no local ID) and "rooms" (other)
  - user-defined "grouping" like discord server grouping, where the user can group chats under tabs without a 'community' structure

please tell me if these ideas already exist, or if i should move this to a new issue, there are currently 4.3k issues here and i dont know how to search for existing ones for my ideas

**Comment by PeterSR** [comment]
Also

- Being able to use @here and @everyone to mention everyone in the room.

**Comment by ShadowJonathan** [comment]
@PeterSR 
maybe `@room`?

**Comment by t3chguy** [comment]
> clicking a user's username makes a mention to their username appear in the chatbox, i expect their profile to pop up

Click on the avatar

**Comment by ShadowJonathan** [comment]
> > clicking a user's username makes a mention to their username appear in the chatbox, i expect their profile to pop up
> 
> Click on the avatar
> 

Even so, that is a discrepancy between user expectancy and actuality (coming from discord: clicking username == profile popup), plus different reactions even though you'll be clicking on the same "thing" (user avatar and username could be considered the same "thing" in my head that way), also the fact that clicking on a user's avatar has a different reaction wasn't noted anywhere, and so I assumed that "clicking on a user in chat" will produce a mention as the law of the land, that is bad UX then, if a user cannot figure out usage from contextual clues if no explicit mention or tutorial is given

**Comment by trosel** [comment]
> But it should be possible, to be completely honest. There is right now a community for german rooms. He is just collecting german rooms. There can be a community for tech-rooms as well. And a german tech-room can be in both communities at the same time. I think thats really good. The thing is, a room 'can' be in multiple communities. So, why should this be messy. You guys are just not used to that. I like it.
> 
> A clan could create a community for the clan with multiple rooms. And the entry-room can be in the "GAME"-community as well, if desired.

If I try to onboard someone to the matrix ecosystem, sending them to the universal room search is the _last_ thing I want to do. I would just give them a simple link to click and instantly join my room.

Who wants this feature? To me, the whole concept of a global registry of rooms and the ability to tag rooms as part of twitter-like "groups" is a solution looking for a problem. All it does is add complexity and there is no discernible benefit. 

Unless matrix is trying to become the next google and build their own index (I wouldn't be surprised if this was a goal lol), I think it's best to keep it simple for the sake of user sanity. 

> 
> And in fact, i like that and this should be kept in my opinion, because its great. You CAN use it this way, but you don't have to. Maybe add an option to make it not possible for others to add this room to a community.

The problem is, the mere existence of an option like this is added complexity and overhead for _everyone_. It affects everyone whether they like it or not. Now you have an additional set of options and UX that make sense to no one except the priest class of matrix users.

**Comment by acoollevel** [comment]
Should the ability to create private communities be added to this list? #12658
It's a pretty important feature in Discord, and I feel like if people see the message "Warning: any person you add to a community will be publicly visible to anyone who knows the community ID" they will be less likely to switch.

**Comment by DonaldTsang** [comment]
@acoollevel the maths on invitations needs to be reworked, but I agree, we need public invites vs private invites and vetting, and then public rooms vs moderation rooms.

**Comment by heylix** [comment]
> > > clicking a user's username makes a mention to their username appear in the chatbox, i expect their profile to pop up
> > 
> > Click on the avatar
> 
> Even so, that is a discrepancy between user expectancy and actuality (**coming from discord**: clicking username == profile popup), plus different reactions even though you'll be clicking on the same "thing" (user avatar and username could be considered the same "thing" in my head that way), also the fact that clicking on a user's avatar has a different reaction **wasn't noted anywhere**, and so I assumed that "clicking on a user in chat" will produce a mention as the law of the land, that is bad UX then, if a user cannot figure out usage from contextual clues if no explicit mention or tutorial is given

So what you're really saying is that documentation and UI hints for onboarding are lacking. The problem is not that it works in a different way, but that users are accustomed to that. For me it's more intuitive that if I click on a name it appears in the chat as mention, as it worked in IRC.

For me it's logical: If you want to know more about a user, click on the user. If you want to talk to the user, click on their name. It's analogous to the real life: You have to know the name, take it from your head or their name tag into your mouth and mention it when you want to talk to someone, but to really know someone the name is not important, it's rather the face or likeness on which we recognise people. You wouldn't be able to someone if all you had was their face as people are conditioned to only feel mentioned when you say their name.

I think the current behaviour is fine. If it's not, there's a possibility for someone to develop a Discord "skin" or "theme" for Riot which works differently – but I think the standard behaviour shouldn't be changed. Consistency in user experience is important and if we'd change things without thinking about the implications for current users, just because Discord or Slack are doing it differently, we'll be in a bad place.

"Good UX" is not "doing everything like other apps". If it were, we wouldn't have any evolution of UX.

**Comment by DonaldTsang** [comment]
@heylix base and experimentation would be good, and treat Discord as the "gold" standard while we try to mine for platinum.

**Comment by Kellegram** [comment]
Agreed. Copy what's good, improve on what's subpar. No need to reinvent the wheel, Discord does a lot of things good from UI standpoint, but also does many things poorly, which is unsolvable by it being closed-source, centralized and catering to the folk that are indifferent as long as it hand holds them enough.

**Comment by MyriaCore** [comment]
As a meta-issue, maybe it'd make sense to update the issue description with
related issues for all of the checkboxes? It'd probably lead to much more
productive discussion, as there would be a summarized single-source-of-truth,
and people wouldn't need to go hunting in the comments for the current state of
these topics.

In general, we also definitely shouldn't be relying on this issue for
discussion.  The way I see it, all of these topics should be given sub-issues of
their own in the relevant channels so they can be implemented more quickly.

I've only done a bit of searching, but here are some of my findings for existing
related proposals:

* [[MSC2545](https://github.com/matrix-org/matrix-doc/pull/2545)] proposes
  custom emojis (<img> tags that render as emoticons), and associated
  functionality (such as making them room-based or community-based)
* Community support is weird, since communities in matrix are a bit jankey, and
  don't *really* do what they appear to do. This is an aside but the issues with
  communities can very likely be alleviated through
  [[MSC1772](https://github.com/matrix-org/matrix-doc/pull/1772)], which
  proposes "spaces", or "groups as rooms". Related is the proposal here [to
  support [MSC1772]](https://github.com/vector-im/element-web/issues/15930).
  For those who don't wanna read the full proposal,
  [[MSC2946](https://github.com/matrix-org/matrix-doc/pull/2946)] does a good
  job of summarizing it.
* Audio Chat is mentioned in matrix-org/matrix-doc#2851, but it's mentioned as
  an in-house thing, rather than a fuller third-party integration thing
  (e.g. via mumble or jitsi). A completely unrelated implementation of Voice
  channels can be seen [over in the matrix-react-sdk
  project](https://github.com/matrix-org/matrix-react-sdk/pull/4400). AFAIK this
  uses jitsi to implement voice channels. For sure show your support over there
  cause I think that PR is on the money. 
* Wasn't mentioned in the issue desc but
  [[MSC2199](https://github.com/matrix-org/matrix-doc/pull/2199)] proposes
  first-class DMs

I'm super new to the matrix ecosystem, so I personally find it difficult to
decide where to post proposals / issues often. Here's what I've sussed out so
far:

* Details proposals for new features that tie into matrix's SDK should probably
  go in [matrix-doc](https://github.com/matrix-org/matrix-doc), as most of the
  features that are implemented in the matrix sdk's seem to come from these
  MSC's.
* UI proposals that would be probably need to be available to all matrix
  webclients should go in
  [matrix-org/matrix-react-sdk](https://github.com/matrix-org/matrix-doc)
* Recommendations for the element.io webapp or the element.io desktop app should
  go here, in [element-web](https://github.com/vector-im/element-web)

**Comment by MyriaCore** [comment]
@DonaldTsang would you mind changing the issue description as discussed above?

<details>
<summary>atm you have this:</summary>

> * [ ]  Room feature support
>   * [ ]  Custom server-based and room-based emojis
>     * Have custom server emojis downloaded and be usable as a basic feature
>   * [ ]  Emoji collection sharing
>     * Similar to Telegram, Messenger, LINE and Whatsapp, but shareable and if possible, self-hosted or user-hosted
>   * [ ]  Emoji reaction for texts and files (useful for voting)
>   * [ ]  Automated (non-manual) key sharing
>     * Some people are sick of needing to sharing key every time someone gets in the room, they would like to have auto key sharing once in an E2E room
>   * [ ]  Multi-room search
> * [ ]  Community support
>   * [x]  Community id generation
>   * [ ]  Community room mass generation
>   * [x]  Community room searching
>   * [ ]  Community multi-admin multi-mod support
>   * [ ]  Community and sub-community sort
> * [ ]  Role and access support
>   * [ ]  Mass room invitation by role or community
>     * invitation to a server once application is approved
>   * [ ]  Mass room banning/kicking by role or community
>     * for when "unfavorable" behavior by certain users happen
>   * [ ]  Community room sub-categorization and access
>     * for visitor vs club member vs club staff separation
>   * [ ]  Role application and removal
>     * for upgrading users to automatically bring them to certain rooms
>   * [ ]  Announcement rooms by role
>     * rooms that allows viewing but not posting unless it is staff
>   * [ ]  Reaction allowance by role
>     * for voting by certain role but not others
> * [ ]  audio chat support
>   * [ ]  self-host SIP room support (tying SIP with server)
>   * [ ]  self-host  Mumble room support (tying Mumble with server)
>   * [ ]  Radio/Playlist bot support
> * [ ]  collaborative support
>   * [ ]  self-host Etherpad/Firepad support (tying *pad with server)
>   * [ ]  self-host Hackmd support (tying Hackmd with server)
</details>



<details>
<summary>But as of the time of writing, I feel this would be more appropriate:</summary>

* [ ]  Room feature support
  * [ ]  Custom server-based and room-based emojis
    * Proposed by [[MSC2545](https://github.com/matrix-org/matrix-doc/pull/2545)]
  * [ ]  Emoji collection sharing
    * Similar to Telegram, Messenger, LINE and Whatsapp, but shareable and if possible, self-hosted or user-hosted
  * [x]  Emoji reaction for texts and files (useful for voting)
  * [ ]  Automated (non-manual) key sharing
    * Some people are sick of needing to sharing key every time someone gets in the room, they would like to have auto key sharing once in an E2E room
  * [ ]  Multi-room search
* [ ]  Community support
  * Element's "Communities" are actually an undocumented matrix feature called [groups](https://github.com/matrix-org/matrix-doc/issues/971), which happens to use a quite [jankey API](https://github.com/matrix-org/matrix-doc/blob/ff85e61be908f4f5f0ba846da0dd6c439147a1b0/proposals/1772-groups-as-rooms.md#appendix-problems-with-the-r0groups-api) [[MSC1772](https://github.com/matrix-org/matrix-doc/pull/1772)] proposes that the we refactor the jank a bit to make groups easier to work with.
  * [This issue](https://github.com/vector-im/element-web/issues/15930) proposes that element.io's client should support MSC1772, instead of using the undocumented groups API
  * [x]  Community id generation
  * [ ]  Community room mass generation
  * [x]  Community room searching
  * [ ]  Community multi-admin multi-mod support
  * [ ]  Community and sub-community sort
* [ ]  Role and access support
  * [ ]  Mass room invitation by role or community
    * invitation to a server once application is approved
  * [ ]  Mass room banning/kicking by role or community
    * for when "unfavorable" behavior by certain users happen
  * [ ]  Community room sub-categorization and access
    * for visitor vs club member vs club staff separation
  * [ ]  Role application and removal
    * for upgrading users to automatically bring them to certain rooms
  * [ ]  Announcement rooms by role
    * rooms that allows viewing but not posting unless it is staff
  * [ ]  Reaction allowance by role
    * for voting by certain role but not others
* [ ]  audio chat support
  * See matrix-org/matrix-doc#2903 for first-class "voice/video rooms".
  * See [this Proof of Concept implementation](https://github.com/matrix-org/matrix-react-sdk/pull/4400) for voice room / voice channel UI in the matrix react sdk
  * [ ]  self-host SIP room support (tying SIP with server)
  * [ ]  self-host  Mumble room support (tying Mumble with server)
  * [ ]  Radio/Playlist bot support
* [ ]  collaborative support
  * [ ]  self-host Etherpad/Firepad support (tying *pad with server)
  * [ ]  self-host Hackmd support (tying Hackmd with server)
</details>

**Comment by ShadowJonathan** [comment]
@MyriaCore I have the feeling that @DonaldTsang has been online for a while for some good reason, I don't think he'll have time to edit the original description anytime soon, so a member of the repo has to do that.

**Comment by DonaldTsang** [comment]
@ShadowJonathan apologies, I would not mind doing it, however if it is going to be changed often, the dev team should change it instead.

**Comment by erkinalp** [comment]
The relevant MSC is https://github.com/matrix-org/matrix-doc/pull/3052.

**Comment by xekon** [comment]
> Mumble



> > > self-host Mumble room support (tying Mumble with server)
> 
> Nothing stops you from hosting a mumble server and using the mumble web UI as a widget.

I am interested in having a voice chat room on my matrix server. It sounds as though mumble can facilitate this?

If so would it show up as a room in element?

I am not afraid of trying something a little outside of the box if the end result is being able to jump into a voice room and immediately being able to voice listen or voice talk with others, without the need to be invited into a call, etc.

I would then be able to pull some friends from discord and onto my matrix server.

I have setup a mumble server before on my linux server, i have just never tried to integrate mumble with matrix.

**Comment by SimonBrandner** [comment]
@xekon, that won't be that easy. You would need something like https://github.com/matrix-org/matrix-react-sdk/pull/4400 which is probably terribly outdated at the moment. You would essentially need to re-write that PR for Mumble and then probably maintain your fork because I have my doubts about this getting merged upstream. Upstream plans Matrix-based voice/video rooms at some point, so this would be an effort that would lead to something that would get replaced eventually. Though now that I think of it, I've heard of an Element fork that already does this. I'd try asking in the [Element Web/Desktop](https://matrix.to/#/#element-web:matrix.org) room maybe someone there can give you directions. In general this question probably should be there rather than here, since this is an issue tracker, not a support forum.

**Comment by ata-star** [comment]
msc2545 implementation would be nice

**Comment by Jieiku** [comment]
> msc2545 implementation would be nice

What is msc2545?

**Comment by natrius** [comment]
I think https://github.com/matrix-org/matrix-spec-proposals/pull/2545 is meant. On the other hand, mumble is currently rewriting its core (i think) and it should be easier to implement this afterwards. https://github.com/mumble-voip/ 
Voice-Channels would should have priority above sticker packs :)

**Comment by prabhathc** [comment]
Hey all, is it possible to also consider the deafen feature? Deafen mutes the user's audio output and mutes all incoming audio from group calls just for the deafened user, convenient feature that should be straightforward to add.

**Comment by heyakyra** [comment]
https://github.com/vector-im/element-call/issues/223 and https://github.com/vector-im/element-call/issues/174 would be nice extensions to the now-merged https://github.com/matrix-org/matrix-spec/issues/730

**Comment by slimsag** [comment]
Could "Increase vertical density of new room list #14099" be added to this meta-issue? I think this is one big drawback compared to Discord's UX right now.

Compare the visual density of Element:

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/3173176/177220414-ae5b56cc-a304-45d1-aeba-5b1b70e38f09.png">

vs. a similar Discord channel organization capabilities:

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/3173176/177220443-697bcc49-8f67-4254-b705-20cd28869d5d.png">

**Comment by erkinalp** [comment]
Well, Discord is also adding a few ideas from Element, for example, the ability to selectively hide guild channels.

**Comment by kittykat** [comment]
Moving this issue to discussions in Element meta as we need to make a cross platform decision on how to proceed :+1:

**Comment by 42Willow** [comment]
I personally prefer the way element is structured
